### PR TITLE
[Modules] Added Private Endpoint support to Cosmos DB

### PR DESCRIPTION
**Pull request by AlexanderSehr**
# Description

- Added Private Endpoint support to Cosmos DB
- Added test case in SQL DB template file

| Pipeline |
| - |
| [DocumentDB - DatabaseAccounts](https://github.com/Azure/ResourceModules/actions/workflows/ms.documentdb.databaseaccounts.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: docs/wiki/The library - Module design.md [hunk 1]
```diff
@@ -48,10 +48,12 @@ They can be deployed in different configurations just by changing the input para
...
   1. **Diagnostic logs** and **metrics** (you can have them sent to one ore more of the following destination types: storage account, log analytics and event hub).
-  2. Resource and child resource level **RBAC** (for example, providing data contributor access on a storage account; granting file share/blob container level access in a storage account)
-  3. **Tags** (as objects)
-  4. **Locks**
-  5. **Private Endpoints** (if supported)
+  1. Resource and child-resource level **RBAC** (for example, providing data contributor access on a storage account; granting file share/blob container level access in a storage account)
+  1. **Tags** (as objects)
+  1. **Locks**
+  1. **Private Endpoints** (if supported)
+  1. **User-defined Types** (if supported)
```
Comment (eriqua) [inline docs/wiki/The library - Module design.md:55]: 🪄

Comment (AlexanderSehr) [reply docs/wiki/The library - Module design.md:55]: No, it seems I used the wrong branch as a basis. Will undo.

Comment (AlexanderSehr) [reply docs/wiki/The library - Module design.md]: @eriqua, removed

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**